### PR TITLE
feat(developer): escape bad markers 🙀 

### PR DESCRIPTION
**Pull request by srl295**
- escape markers in regex (`\u0001` format) to prevent the n'th marker from having some other meaning! 
- with test case

#9121

@keymanapp-test-bot skip

### CODE REVIEW

File: core/tests/unit/ldml/keyboards/k_211_marker_escape.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+<!DOCTYPE keyboard3 SYSTEM "../../../../../resources/standards-data/ldml-keyboards/techpreview/dtd/ldmlKeyboard3.dtd">
+<keyboard3 locale="en" conformsTo="techpreview">
+  <info author="srl295" indicator="🙀" layout="qwerty"  name="Marker Escape Test"/>
+
+  <displays>
+  </displays>
+
+  <keys>
+    <!-- fourty markers.  so that the next one would have its representation as '(' -->
+    <key id="1" output="\m{m01}\m{m02}\m{m03}\m{m04}\m{m05}\m{m06}\m{m07}\m{m08}\m{m09}\m{m10}\m{m11}\m{m12}\m{m13}\m{m14}\m{m15}\m{m16}\m{m17}\m{m18}\m{m19}\m{m20}\m{m21}\m{m22}\m{m23}\m{m24}\m{m25}\m{m26}\m{m27}\m{m28}\m{m29}\m{m30}\m{m31}\m{m32}\m{m33}\m{m34}\m{m35}\m{m36}\m{m37}\m{m38}\m{m39}\m{m40}"/>
```
Comment (srl295) [inline core/tests/unit/ldml/keyboards/k_211_marker_escape.xml:16]: 🙀

**Comment by srl295** [comment]
interesting. 

So, 

```c
  normalize_nfd_markers(patstr);
```

Doesn't work anymore, because it expects the sentinel pattern, not:  U+FFFF U+0008 `\uNNNN`

So I need a new kind of 'normalize_nfd_markers' (or a switch) that works w/ regex pattern space.

**Comment by srl295** [comment]
the parse/format could be more optimized. or maybe use something from ICU.

**Review by mcdurdin — APPROVED** [review]

File: core/src/ldml/ldml_transforms.cpp [hunk 2]
```diff
@@ -950,17 +950,17 @@ bool normalize_nfd(std::u16string &str) {
   return normalize(nfd, str, status);
 }
 
-bool normalize_nfd_markers(std::u16string &str, marker_map &map) {
+bool normalize_nfd_markers(std::u16string &str, marker_map &map, bool for_regex) {
```
[the remark this answers is not in the document]
Comment (srl295) [reply core/src/ldml/ldml_transforms.cpp]: done

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.238-alpha](https://keyman.com/downloads/releases/alpha/17.0.238)

**Pull request merged**